### ISSUE TITLE
Consider forbidding <frame> element from entering fullscreen

### ISSUE THREAD

**Issue by upsuper** [whatwg/fullscreen#112]
It seems that `<frame>` element is not allowed to be out-of-flow, so it's incompatible with the top layer model, like internal SVG and MathML elements. And it is a deprecated tech anyway.

I propose that we add `<frame>` element to the same place where we rule out `<dialog>` from entering fullscreen.

(This was raised from [a fuzzy bug which triggers an internal assertion in Gecko](https://bugzilla.mozilla.org/show_bug.cgi?id=1415166).)

**Comment by foolip** [comment]
Fullscreen inside of `<frame>` used to work in Chromium. But here it's `frame.requestFullscreen()` so not exactly the situation. Disallowing this would be straightforward and fine I think, unlikely to be a web compat problem. Care to send spec+test PRs?

**Comment by upsuper** [comment]
Fullscreen inside of `<frame>` should have been forbidden by the "allowed to use" algorithm anyway. Also `<frame>` being able to enter fullscreen is the prerequisite for its content to do so, like `<iframe>`.

It is not clear to me how would fullscreen be applied for `<frame>` given that it cannot go out-of-flow...

I can do the spec+test... when I have time at some point...

**Comment by foolip** [comment]
> Fullscreen inside of `<frame>` should have been forbidden by the "allowed to use" algorithm anyway.

Right, but it [used to work](https://chromium.googlesource.com/chromium/src/+/32a059337eb1c538242091adc170c41be325443b), is all, which just means that whatever the problem is in Gecko probably doesn't apply to Chromium. I'm still fine with not allowing it.

**Comment by annevk** [comment]
Does this also apply to `frameset`?
